### ISSUE TITLE
[java] ConsecutiveLiteralAppends false positive over catch

### ISSUE THREAD

**Issue by vbrandl** [pmd/pmd#1632]
**Affects PMD Version:** 6.11.0

**Rule:** [ConsecutiveLiteralAppends](https://pmd.github.io/pmd-6.11.0/pmd_rules_java_performance.html#consecutiveliteralappends)

**Description:** When appending a literal before a try/catch block and also inside the try block, after calling a method, that might throw a exception, PMD falsely reports a `ConsecutiveLiteralAppends` lint.

**Code Sample demonstrating the issue:**

```java
public final class Test {
    public String foo() {
        final StringBuilder sb = new StringBuilder();
        sb.append("literal1");
        try {
            final String res = bar();
            sb.append("literal2").append(res);
        } catch (Exception e) {
            sb.append("literal3");
        }
        return sb.toString();
    }

    public String bar() throws Exception {
        throw new Exception();
    }
}
```

**Running PMD through:** *Gradle*

**Comment by Shubham-2k17** [comment]
```
public final class Test {
    public String foo() {
        final StringBuilder sb = new StringBuilder();
        sb.append("literal1");
        sb.append("literal2" + "literal3");
        return sb.toString();
    }
} 
```
is not being handled by ConsecutiveLiteralAppendsRule.
I think this is **false negative**. I think the expression inside append needs to be evaluated for solving all the problems.

**Comment by vbrandl** [comment]
As far as I know, concatenation of string literals will be handled at compile time. So `"literal2" + "literal3"` should be allowed. I think it is even the other way and PMD warns, if you call `append` multiple times on string literals.
